### PR TITLE
adding CBP by default

### PR DESCRIPTION
**Pull request by cryptomail**
Fixes issue https://github.com/blakmatrix/node-zendesk/issues/354
Adds examplests too 😸 (example tests)
@blakmatrix Please review :)

### CODE REVIEW

File: examples/check-auth-promise.js [hunk 1]
```diff
@@ -2,9 +2,9 @@ var exampleConfig = require('./exampleConfig');
 var zd = require('../lib/client');
 
 var client = zd.createClient({
-  username:  exampleConfig.auth.username,
-  token:     exampleConfig.auth.token,
-  remoteUri: exampleConfig.auth.remoteUri
+    username:  process.env.ZENDESK_TEST_USERNAME || exampleConfig.auth.username,
+    token:     process.env.ZENDESK_TEST_TOKEN || exampleConfig.auth.token,
+    remoteUri: process.env.ZENDESK_TEST_REMOTEURI || exampleConfig.auth.remoteUri
```
Comment (blakmatrix) [inline examples/check-auth-promise.js:7]: Awesome Thanks for adding this!

File: lib/client/client.js [hunk 2]
```diff
@@ -170,7 +170,11 @@ Client.prototype.requestAll = function (method, uri) {
         responseList.push(response);
         resultList.push(result);
       }
-      nextPage = result ? result.next_page : null;
+      if (result && result.links && result.links.next) {
+        nextPage = result.links.next;
+      } else {
+        nextPage = result ? result.next_page : null;
+      }
```
Comment (blakmatrix) [inline lib/client/client.js:177]: Very Nice! Excellent updates @cryptomail !

**Review by blakmatrix — APPROVED** [review]
Awesome! Thanks for these changes!

**Comment by cryptomail** [comment]
@blakmatrix <3 Let's work together maybe to modernize this library just a smidge <3.
More than happy to coordinate more work with you.

**Pull request merged**